control-plane/api/v1alpha1/registration_webhook_test.go

```diff
@@ -0,0 +1,330 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package v1alpha1
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	logrtest "github.com/go-logr/logr/testr"
+	"github.com/stretchr/testify/require"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func TestValidateRegistration(t *testing.T) {
+	cases := map[string]struct {
+		newResource   *Registration
+		expAllow      bool
```

Comment:
what does exp in expAllow mean? expect?

Reply:
`expectedAllow` I can expand this (I copied/pasted from another webhook test)